### ISSUE TITLE
Rethink inline javascript for reports for JupyterLab compatibility

### ISSUE THREAD

**Issue by sserita** [pyGSTio/pyGSTi#205]
**Is your feature request related to a problem? Please describe.**
The current report workspaces heavily uses inline javascript for its interactive rendering. This works great in regular Jupyter notebooks, but JupyterLab silently suppresses the JS execution. If more users are moving towards JupyterLab, this is something we should provide an alternative for.

**Describe the solution you'd like**
Maintain the same level of display functionality as classic Jupyter notebooks in JupyterLab.

**Describe alternatives you've considered**
- Try to find an extension/setting that reenables the current JS execution in JupyterLab, and require it for interactive report generation.
- Rework the JS execution into something that is natively enabled in JupyterLab.

Not sure which is the easier solution yet. The former seems like less work in the PyGSTi codebase but may not be easy to find/require. More likely, we should make sure to keep this in mind the next time we spend significant effort on upgrading the reports.

**Additional context**
I have not searched exhaustively yet, but this thread is relevant: https://github.com/jupyterlab/jupyterlab/issues/3118

**Comment by rileyjmurray** [comment]
Update on this. I haven't been able to get notebook reporting to work even if I try to avoid jupyterlab. Here's what I see when I install nbclassic:

![image](https://github.com/user-attachments/assets/78af3497-f6b5-4281-87cf-15dde5c70962)

There might still be a way to avoid this. I'm just not super familiar with the python notebook ecosystem. In fact, even when _only_ installing nbclassic from pip, I see the following when launching the server. Note that it mentions JupyterLab being loaded and linked.
<img width="1058" alt="image" src="https://github.com/user-attachments/assets/58536417-0c92-4a29-bd4f-0e4688c9d8aa" />

**Comment by sserita** [comment]
Yes, I have recently been getting reports of this breaking even if avoiding JupyterLab. However, there are some known workarounds by reaching directly into the plotly objects to pull out the plots - I wonder if we can turn that into less of a hack (or at least a canonicalized hack) and finally close this issue. I'll put this on our agenda to discuss the next time we do issue re-triage (which might be next week).

**Comment by sserita** [comment]
This might need to be split into two: short-term fix for report notebooks, and long-term fix to handle more complicated things like switchboards.
